### PR TITLE
FIS: Additional FIRInstallationsItem validation

### PR DESCRIPTION
**Pull request by maksymmalyhin**
Related to #6563. 

- more detailed validation of `FIRInstallationsItem` read from storage to catch a potential data corruption (e.g. manual modifications of Keychain item on macOS, etc.)
- validation of `FIRInstallationsItem` before sending registration request.

### CODE REVIEW

**Comment by maksymmalyhin** [comment]
cc @andirayo

**Review by paulb777 — APPROVED** [review]

File: FirebaseInstallations/Source/Tests/Unit/FIRInstallationsIDControllerTests.m [hunk 1]
```diff
@@ -1504,4 +1446,76 @@ - (void)expectBackoffEvent:(FIRInstallationsBackoffEvent)event
   OCMVerifyAll(self.mockBackoffController);
 }
 
+- (void)assertGetInstallation_NewFIDCreatedAndRegistered {
```
Comment (chliangGoogle) [inline FirebaseInstallations/Source/Tests/Unit/FIRInstallationsIDControllerTests.m:1449]: nit:should method name be camel case without "_"?

**Review by chliangGoogle — APPROVED** [review]

File: FirebaseInstallations/Source/Tests/Unit/FIRInstallationsIDControllerTests.m (hunk 1 above)
Comment (maksymmalyhin) [reply FirebaseInstallations/Source/Tests/Unit/FIRInstallationsIDControllerTests.m:1449]: I added "_" in some tests to the very long names to break the name to logical part to improve readability. I struggled to come up with anything better.

Comment (chliangGoogle) [reply FirebaseInstallations/Source/Tests/Unit/FIRInstallationsIDControllerTests.m:1449]: That's probably fine. Just want to call out the style guide. 
I'm okay if it leaves like this.

**Pull request merged**